mantidproject/mantid
# Muon fitting panes

**Description of work.**
This PR adds the new plot fitting panes to muon analysis and frequency domain analysis. I had to do both as the changes meant it was not possible to do one and not the other. This new plotting should make it significantly easier to add new plotting functionality. 




**To test:**


Open muon analysis
Load some data
Make sure it can tile data correctly (by run and group/pair)
Go to fitting (the plot should update)
Make sure the plot matches the data being fitted (including simultaneous) 
Check plot guess works
Do a fit
Check plot diff works
Check sequential fitting works

Open frequency domain analysis
Load some data
Do some transforms (FFT and maxent)
Check they are plotted correctly (will always default to the modulus)
Go to fitting and make sure that the displayed data matches the selection
Check plot guess
Check doing a fit
Check plot diff works 

Fixes #31495 and #31496. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
